src/CommunityToolkit.Aspire.Hosting.SqlDatabaseProjects/SqlProjectBuilderExtensions.cs

```diff
@@ -0,0 +1,110 @@
+using Aspire.Hosting.ApplicationModel;
+using Microsoft.Build.Locator;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using CommunityToolkit.Aspire.Hosting.SqlDatabaseProjects;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for adding SQL Server Database Projects to the application.
+/// </summary>
+public static class SqlProjectBuilderExtensions
+{
+    /// <summary>
+    /// Static constructor to ensure that MSBuild assemblies are properly loaded.
+    /// </summary>
+    static SqlProjectBuilderExtensions()
+    {
+        if (!MSBuildLocator.IsRegistered)
+        {
+            MSBuildLocator.RegisterDefaults();
+        }
+    }
+
+    /// <summary>
+    /// Adds a SQL Server Database Project resource to the application based on a referenced MSBuild.Sdk.SqlProj project.
+    /// </summary>
+    /// <typeparam name="TProject">Type that represents the project that produces the .dacpac file.</typeparam>
+    /// <param name="builder">An <see cref="IDistributedApplicationBuilder"/> instance to add the SQL Server Database project to.</param>
+    /// <param name="name">Name of the resource.</param>
+    /// <returns>An <see cref="IResourceBuilder{T}"/> that can be used to further customize the resource.</returns>
+    public static IResourceBuilder<SqlProjectResource> AddSqlProject<TProject>(this IDistributedApplicationBuilder builder, string name)
+        where TProject : IProjectMetadata, new()
+    {
```

Comment:
Wouldn't we want to do a `ThrowIfNullOrWhitespace` on the `name` attribute?

Reply:
No, and we probably don't even need to do the `ArgumentNullException` on the `name` param (but let's leave it for now), because the analyser will result in a compiler error (`ASPIRE006`) if your supplied name doesn't meet the requirements (ASCII letters, digits, and hyphens, greater than 1 character length and less than 64).